program/src/processor.rs

```diff
@@ -0,0 +1,485 @@
+//! Program processor
+
+use {
+    crate::{
+        check_id,
+        instruction::ProgramInstruction,
+        state::{
+            AddressLookupTable, LookupTableStatus, ProgramState, LOOKUP_TABLE_MAX_ADDRESSES,
+            LOOKUP_TABLE_META_SIZE,
+        },
+    },
+    solana_program::{
+        account_info::{next_account_info, AccountInfo},
+        clock::{Clock, Slot},
+        entrypoint::ProgramResult,
+        msg,
+        program::{invoke, invoke_signed},
+        program_error::ProgramError,
+        pubkey::{Pubkey, PUBKEY_BYTES},
+        rent::Rent,
+        slot_hashes::MAX_ENTRIES,
+        system_instruction,
+        sysvar::Sysvar,
+    },
+};
+
+// [Core BPF]: Locally-implemented
+// `solana_sdk::program_utils::limited_deserialize`.
+fn limited_deserialize<T>(input: &[u8]) -> Result<T, ProgramError>
+where
+    T: serde::de::DeserializeOwned,
+{
+    solana_program::program_utils::limited_deserialize(
+        input, 1232, // [Core BPF]: See `solana_sdk::packet::PACKET_DATA_SIZE`
+    )
+    .map_err(|_| ProgramError::InvalidInstructionData)
+}
```

Comment:
This is fine for the initial PR, but we should consider eventually removing this for performance reasons

Reply:
Circling back on this after [expanding on the extend table tests](https://github.com/solana-program/address-lookup-table/pull/7#discussion_r1516919049), the `1232` cap is intentional, to limit the number of `new_addresses` that could be provided.

Maybe we just roll it into an `unpack` function for `AddressLookupTableInstruction`?

Reply:
Oh sorry, I just meant that we should use an in-place deserializer for the instructions rather than using bincode since it's very CU heavy.

Is there a test to see if it's possible to add the max possible addresses in one BPF instruction? At 1232 bytes max... and 64 bytes for a signature, 32 for the payer, 32 for the address lookup table, and 32 for the program id, that gives us around 32-33 max new addresses that can be added at once? As long as we can do that in 200k compute units, we should be good for now